README.md

```diff
@@ -1,3 +1,607 @@
-# @octokit/webhooks.js
+# @octokit/webhooks
 
-> GitHub webhooks toolset
+> GitHub webhooks toolset for Node.js
+
+[GitHub webhooks](https://developer.github.com/webhooks/) can be registered in multiple ways
+
+1. In repository or organization settings on [github.com](https://github.com/).
+2. Using the REST API for [repositories](https://developer.github.com/v3/repos/hooks/) or [organizations](https://developer.github.com/v3/orgs/hooks/)
+3. By installing a [GitHub App](https://developer.github.com/apps/).
+
+`@octokit/webhooks` helps you handle the webhooks sent by GitHub to your server.
+
+## Example
+
+```js
+// install with: npm install @octokit/webhooks
+const Webhooks = require('@octokit/webhooks')
+const receiver = Webhooks.receiver({secret: 'mysecret'})
+
+// called with installation webhook payload
+// see https://developer.github.com/v3/activity/events/types/#installationevent
+receiver.on('installation.deleted', {installation, sender} => {
+  console.log(`${sender.login} deleted installation for ${installation.account.login}`)
+  // logs e.g. "octocat deleted installation for octocat"
```

Comment:
The current implementation would only pass the event payload to the event handlers, while the `github-webhook-handler` module emits an object with the properties `{event, id, payload, protocol, host, url}`

https://github.com/octokit/webhooks.js/blob/9e0d6364e0e3e0b07fc51237e25bdffafdd9174d/github-webhook-handler.js#L103-L110

I don’t see a use case for protocol, host or url, but I think it does make sense to pass the event name, request id and the payload as separate properties.

Reply:
I can see a use-case for usage with GitHub Enterprise to provide the `host` or `url`, if only to be able to explicitly verify that webhooks are coming from the correct GHE url. Don't know about `protocol` though.

Reply:
This would only be relevant if a `secret` would not  be configured, right? I can imagine a case where the same webhook URL would handle events from both github.com and GHE and then distinguish them based on the host. But if we were to enforce the usage of `secret` – which [I think we should](https://github.com/gr2m/octokit-webhooks/pull/1#issuecomment-345381573) – this becomes a non issue, right?

I’d like to make the decision on wether to enforce `secret` or not first. I had a discussion with [Carter](https://twitter.com/crtr0) the other day at Offline Camp. He said that one of the biggest trouble they had was to explain what the `secret` is and why it’s important. Folks would build webhook handlers ignoring it. I think one way to avoid this is to provide tooling that enforces the best practise. Unless someone has a good use case for why it should not

Reply:
Totally fair - the use of secrets is a valid point of why my proposed use case isn't right. However, new use case: if my app writes comments that link to other issues or repos, those URLs will need to have the host associated so that one app can be installed against both GHE _and_ GitHub.com, but still point URLs to the appropriate resource.

Reply:
The `host`, `protocal` and `url` are not for the GitHub instance sending the webhook, but for the receiving webhook event server. They only thing this could be helpful is if you have the same server instance receiving events from different GitHub instances at different domains, I think? 

I think your use case is interesting though, for that to work we would need a referer header though. And ideally a header which tells us the GHE version ..., maybe let’s create a follow up issue for that and raise it with the GitHub platform folks?

Reply:
> The host, protocal and url are not for the GitHub instance sending the webhook, but for the receiving webhook event server.

Oooh I see what you mean, that makes sense. Sorry about the confusion.

> a referer header though

That's a good idea.

Reply:
created a follow up issue to get back to it later: https://github.com/gr2m/octokit-webhooks/issues/2